ush/python/pygfs/task/land_analysis.py

```diff
@@ -118,11 +110,18 @@ def _gtsbufr2iodax(exe, yaml_file):
             except Exception:
                 raise WorkflowException(f"An error occured during execution of {exe} {yaml_file}")
 
-        # execute BUFR2IODAX to convert adpsfc bufr data into IODA format
-        _gtsbufr2iodax(exe, os.path.join(localconf.DATA, "bufr_adpsfc_snow.yaml"))
+        # Loop over entries in prep_gts_config.bufr2ioda keys
+        # 1. generate bufr2ioda YAML files
+        # 2. execute bufr2ioda.x
+        for name in prep_gts_config.bufr2ioda.keys():
+            gts_yaml = os.path.join(self.runtime_config.DATA, f"bufr_{name}_snow.yaml")
+            logger.info(f"Generate BUFR2IODA YAML file: {gts_yaml}")
+            temp_yaml = parse_j2yaml(prep_gts_config.bufr2ioda[name], localconf)
+            save_as_yaml(temp_yaml, gts_yaml)
+            logger.info(f"Wrote bufr2ioda YAML to: {gts_yaml}")
 
-        # execute BUFR2IODAX to convert snocvr bufr data into IODA format
-        _gtsbufr2iodax(exe, os.path.join(localconf.DATA, "bufr_snocvr_snow.yaml"))
+            # execute BUFR2IODAX to convert {name} bufr data into IODA format
+            _gtsbufr2iodax(exe, os.path.join(localconf.DATA, f"bufr_{name}_snow.yaml"))
```

Comment:
```suggestion
            _gtsbufr2iodax(exe, gts_yaml)
```
In my second (revised diff.txt), I removed the need to construct this again.

Reply:
Done the changes.